code/datums/components/item_two_handed.dm

```diff
@@ -0,0 +1,339 @@
+ * * force_unwielded (optional) The force setting when the item is unwielded, do not use with force_multiplier
+ * * icon_prefix (optional) The prefix of the items icon to be used with wielded status
+ * * icon_update_callback (optional) proc (wielded) Callback with wielded status
+ * * on_wield_callback (optional) proc (user) Callback on wield of the item
+ * * on_unwield_callback (optional) proc (user) Callback on unwield of the item
+ */
+/datum/component/two_handed/Initialize(require_twohands=FALSE, wieldsound=FALSE, unwieldsound=FALSE, attacksound=FALSE, \
+										force_multiplier=0, force_wielded=0, force_unwielded=0, icon_prefix=FALSE, \
+										datum/callback/icon_update_callback=null, datum/callback/on_wield_callback=null, datum/callback/on_unwield_callback=null)
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.require_twohands = require_twohands
+	src.wieldsound = wieldsound
+	src.unwieldsound = unwieldsound
+	src.attacksound = attacksound
+	src.force_multiplier = force_multiplier
+	src.force_wielded = force_wielded
+	src.force_unwielded = force_unwielded
+	src.icon_prefix = icon_prefix
+	src.icon_update_callback = icon_update_callback
+	src.on_wield_callback = on_wield_callback
+	src.on_unwield_callback = on_unwield_callback
+
+// Inherit the new values passed to the component
+/datum/component/two_handed/InheritComponent(datum/component/two_handed/new_comp, original, require_twohands, \
+												wieldsound, unwieldsound, force_wielded, force_unwielded, icon_prefix, \
+												datum/callback/icon_update_callback, datum/callback/on_wield_callback, datum/callback/on_unwield_callback)
+	if(!original)
+		return
+	if(require_twohands)
+		src.require_twohands = require_twohands
+	if(wieldsound)
+		src.wieldsound = wieldsound
+	if(unwieldsound)
+		src.unwieldsound = unwieldsound
+	if(attacksound)
+		src.attacksound = attacksound
+	if(force_multiplier)
+		src.force_multiplier = force_multiplier
+	if(force_wielded)
+		src.force_wielded = force_wielded
+	if(force_unwielded)
+		src.force_unwielded = force_unwielded
+	if(icon_prefix)
+		src.icon_prefix = icon_prefix
+	if(icon_update_callback)
+		src.icon_update_callback = icon_update_callback
+	if(on_wield_callback)
+		src.on_wield_callback = on_wield_callback
+	if(on_unwield_callback)
+		src.on_unwield_callback = on_unwield_callback
+
+// register signals withthe parent item
+/datum/component/two_handed/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_ITEM_EQUIPPED, .proc/on_equip)
+	RegisterSignal(parent, COMSIG_ITEM_DROPPED, .proc/on_drop)
+	RegisterSignal(parent, COMSIG_ITEM_ATTACK_SELF, .proc/on_attack_self)
+	RegisterSignal(parent, COMSIG_ITEM_ATTACK, .proc/on_attack)
+	RegisterSignal(parent, COMSIG_ATOM_UPDATE_ICON, .proc/on_update_icon)
+	RegisterSignal(parent, COMSIG_TRY_TWOHANDED_WIELD, .proc/try_wield)
+	RegisterSignal(parent, COMSIG_TRY_TWOHANDED_UNWIELD, .proc/try_unwield)
```

Comment:
You should not be using either of these try signals

Reply:
These signals have been removed.